app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -490,16 +492,20 @@ class GleanMetricsService(private val context: Context) : MetricsService {
         // because it calls Google ad APIs that must be called *off* of the main thread.
         // These two things actually happen in parallel, but that should be ok because Glean
         // can handle events being recorded before it's initialized.
-        gleanInitializer = MainScope().launch {
+        gleanInitializer = MainScope().launch(start = CoroutineStart.LAZY) {
```

Comment:
How much would this delay the call to `Glean.registerPings(Pings)`? Would this still be guaranteed to be called before [this](https://github.com/mozilla-mobile/fenix/blob/0199363577dc654562d9f6c7d7d087ed8fde0014/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L541)? If not, then the `activation` ping will not be sent.

Reply:
Yes, that method would be called later it would still be called in the same order as they are now. In other words, the code in the Job of the `gleanInitializer()` will execute before the code in the Job of the `setStartupMetrics()`. Does that make sense?

Reply:
Of course, if what you are saying is a requirement, then doing what we are currently doing in Fenix is not safe. It is possible that the scheduler swaps these two Jobs and executes `setStartupMetrics` before `gleanInitializer` because we do not specifically order them.

Reply:
Nope

> 
> 
> Of course, if what you are saying is a requirement, then doing what we are currently doing in Fenix is not safe. It is possible that the scheduler swaps these two Jobs and executes `setStartupMetrics` before `gleanInitializer` because we do not specifically order them.

This part should be fine. I'm concerned about Glean.registerPings which should be called before [this](https://github.com/mozilla-mobile/fenix/blob/0199363577dc654562d9f6c7d7d087ed8fde0014/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L541)